pkg/project/filter.go

```diff
@@ -0,0 +1,54 @@
+package project
+
+import (
+	"fmt"
+
+	"github.com/wakatime/wakatime-cli/pkg/heartbeat"
+	"github.com/wakatime/wakatime-cli/pkg/log"
+)
+
+// FilterConfig contains project filtering configurations.
+type FilterConfig struct {
+	// ExcludeUnknownProject determines if heartbeat should be skipped when the project cannot be detected.
+	ExcludeUnknownProject bool
+}
+
+// WithFiltering initializes and returns a heartbeat handle option, which
+// can be used in a heartbeat processing pipeline to filter heartbeats following
+// the provided configurations.
+func WithFiltering(config FilterConfig) heartbeat.HandleOption {
+	return func(next heartbeat.Handle) heartbeat.Handle {
+		return func(hh []heartbeat.Heartbeat) ([]heartbeat.Result, error) {
+			log.Debugln("execute project filtering")
+
+			var filtered []heartbeat.Heartbeat
+
+			for _, h := range hh {
+				err := Filter(h, config)
+				if err != nil {
+					log.Debugln("skipping because of unknown project")
```

Comment:
The name of the function (`Filter`) is very generic. Thus I would not assume at this point, that every error happened due to an empty project. How about using the message from `err.Error()`? In the current implementation this will result in exactly the same log message.

Reply:
That makes sense and I didn't see that `log.Debugln()` was printing the same error message coming from the function. Nice catch!